src/dbnode/storage/database.go

```diff
@@ -1086,6 +1086,41 @@ func (d *db) OwnedNamespaces() ([]databaseNamespace, error) {
 	return d.ownedNamespacesWithLock(), nil
 }
 
+func (d *db) AggregateTiles(
+	ctx context.Context,
+	sourceNsID,
+	targetNsID ident.ID,
+	opts AggregateTilesOptions,
+) (int64, error) {
+	ctx, sp, sampled := ctx.StartSampledTraceSpan(tracepoint.DBAggregateTiles)
+	if sampled {
+		sp.LogFields(
+			opentracinglog.String("sourceNamespace", sourceNsID.String()),
+			opentracinglog.String("targetNamespace", targetNsID.String()),
+			xopentracing.Time("start", opts.Start),
+			xopentracing.Time("end", opts.End),
+			xopentracing.Duration("step", opts.Step),
+		)
+	}
+	defer sp.Finish()
+
+	sourceNs, err := d.namespaceFor(sourceNsID)
+	if err != nil {
+		d.metrics.unknownNamespaceRead.Inc(1)
+		return 0, err
+	}
+
+	targetNs, err := d.namespaceFor(targetNsID)
+	if err != nil {
+		d.metrics.unknownNamespaceRead.Inc(1)
+		return 0, err
+	}
+
+	// TODO: Create and use a dedicated persist manager
```

Comment:
Has this been dealt with in a followup?

Reply:
Yes. In fact, with the new `writer`, we are no longer using cold flushes / perist manager at all.